src/linear_algebra/special_linear_group.lean

```diff
@@ -96,75 +95,70 @@ section coe_lemmas
 
 variables (A B : special_linear_group n R)
 
-@[simp] lemma inv_val : ↑(A⁻¹) = adjugate A := rfl
+@[simp] lemma coe_inv : ↑ₘ(A⁻¹) = adjugate A := rfl
 
-@[simp] lemma inv_apply : ⇑(A⁻¹) = adjugate A := rfl
+@[simp] lemma coe_mul : ↑ₘ(A * B) = ↑ₘA ⬝ ↑ₘB := rfl
 
-@[simp] lemma mul_val : ↑(A * B) = A ⬝ B := rfl
+@[simp] lemma coe_one : ↑ₘ(1 : special_linear_group n R) = (1 : matrix n n R) := rfl
 
-@[simp] lemma mul_apply : ⇑(A * B) = (A ⬝ B) := rfl
+@[simp] lemma det_coe : det ↑ₘA = 1 := A.2
 
-@[simp] lemma one_val : ↑(1 : special_linear_group n R) = (1 : matrix n n R) := rfl
+end coe_lemmas
 
-@[simp] lemma one_apply : ⇑(1 : special_linear_group n R) = (1 : matrix n n R) := rfl
+instance : monoid (special_linear_group n R) :=
+function.injective.monoid coe subtype.coe_injective coe_one coe_mul
 
-@[simp] lemma det_coe_matrix : det A = 1 := A.2
+instance : group (special_linear_group n R) :=
+{ mul_left_inv := λ A, by { ext1, simp [adjugate_mul] },
+  ..special_linear_group.monoid,
+  ..special_linear_group.has_inv }
 
-lemma det_coe_fun : det ⇑A = 1 := A.2
+/-- `matrix.to_lin' A` is a linear equivalence on the special linear group. -/
```

Comment:
This is still in the `namespace special_linear_group`, right?
```suggestion
/-- `special_linear_group.to_lin' A` turns an element of the special linear group into a linear equivalence. -/
```

Reply:
I've reworded slightly differently - let me know if you prefer your spelling.